client/i18n/schema-validation.ts

```diff
@@ -0,0 +1,264 @@
+
+/**
+ * Flattens a nested object structure into a single
+ * object with property chains as keys.
+ * @param {Object} obj Object to flatten
+ * @param {String} namespace Used for property chaining
+ */
+const flattenAnObject = (obj: Record<string, unknown>, namespace = '') => {
+  const flattened: Record<string, unknown> = {};
+  Object.keys(obj).forEach(key => {
+    const value = obj[key];
+    if (Array.isArray(value)) {
+      flattened[namespace ? `${namespace}.${key}` : key] = value;
+    } else if (typeof value === 'object') {
+      Object.assign(
+        flattened,
+        flattenAnObject(
+          value as Record<string, unknown>,
+          namespace ? `${namespace}.${key}` : key
+        )
+      );
+    } else {
+      flattened[namespace ? `${namespace}.${key}` : key] = value;
+    }
+  });
+  return flattened;
+};
+
+/**
+ * Checks if a translation object is missing keys
+ * that are present in the schema.
+ * @param {String[]} file Array of translation object's keys
+ * @param {String[]} schema Array of matching schema's keys
+ * @param {String} path string path to file
+ */
+const findMissingKeys = (file: string[], schema: string[], path: string) => {
+  const missingKeys = [];
+  for (const key of schema) {
+    if (!file.includes(key)) {
+      missingKeys.push(key);
+    }
+  }
+  if (missingKeys.length) {
+    console.warn(
+      `${path} is missing these required keys: ${missingKeys.join(', ')}`
+    );
+  }
+};
+
+/**
+ * Checks if a translation object has extra
+ * keys which are NOT present in the schema.
+ * @param {String[]} file Array of translation object's keys
+ * @param {String[]} schema Array of matching schema's keys
+ * @param {String} path string path to file
+ */
+const findExtraneousKeys = (file: string[], schema: string[], path: string) => {
+  const extraKeys = [];
+  for (const key of file) {
+    if (!schema.includes(key)) {
+      extraKeys.push(key);
+    }
+  }
+  if (extraKeys.length) {
+    console.warn(
+      `${path} has these keys that are not in the schema: ${extraKeys.join(
+        ', '
+      )}`
+    );
+  }
+};
+
+/**
+ * Validates that all values in the object are non-empty. Includes
+ * validation of nested objects.
+ * @param {Object} obj The object to check the values of
+ * @param {String} namespace String for tracking nested properties
+ */
+const noEmptyObjectValues = (
+  obj: Record<string, unknown>,
+  namespace = ''
+): string[] => {
+  const emptyKeys = [];
+  for (const key of Object.keys(obj)) {
+    const value = obj[key];
+    if (Array.isArray(value)) {
+      if (!value.length) {
+        emptyKeys.push(namespace ? `${namespace}.${key}` : key);
+      }
+    } else if (typeof value === 'object') {
+      emptyKeys.push(
+        noEmptyObjectValues(
+          value as Record<string, unknown>,
+          namespace ? `${namespace}.${key}` : key
+        )
+      );
+    } else if (!value) {
+      emptyKeys.push(namespace ? `${namespace}.${key}` : key);
```

Comment:
```suggestion
    const value = obj[key];
    const field = namespace ? `${namespace}.${key}` : key;
    if (Array.isArray(value)) {
      if (!value.length) {
        emptyKeys.push(field);
      }
    } else if (typeof value === 'object') {
      emptyKeys.push(
        noEmptyObjectValues(value as Record<string, unknown>, field));
    } else if (!value) {
      emptyKeys.push(field);
```

Reply:
Thanks for the suggestion😊, I have changed it to this, and linter is also satisfied.